flatdata-cpp/test/GeneratedArchiveTest.cpp

```diff
@@ -330,6 +329,77 @@ TEMPLATE_TEST_CASE_METHOD( Fixture,
+                           "[GeneratedArchive]",
+                           std::true_type,
+                           std::false_type )
+{
+    auto outer_builder = OuterArchiveBuilder::open( Fixture< TestType >::storage );
+    CHECK( outer_builder );
+
+    flatdata::Struct< AStruct > o;
+    ( *o ).value = 17u;
+    outer_builder.set_outer1( *o );
+    outer_builder.set_outer2( *o );
+
+    auto outer = OuterArchive::open( Fixture< TestType >::storage );
+    REQUIRE_FALSE( outer.is_open( ) );
+
+    const char* const expected
+        = R"data(================================================================================
+FATAL: Archive initialization failed. Failed loading mandatory resources.
+================================================================================
+Flatdata Archive: OuterArchive
+================================================================================
+Resource                             Optional  Too Large  Loaded    Details
+================================================================================
+outer1                               NO        NO         YES       Structure of size 1
+outer2                               NO        NO         YES       Structure of size 1
+inner                                NO        NO         NO        FATAL: Resource storage not initialized. Please check archive path.
+|
+|-> Flatdata Archive: InnerArchive
+    inner                            NO        NO         NO        Uninitialized Structure AStruct
+================================================================================
+)data";
+
+    REQUIRE( outer.describe( ) == expected );
+}
+
+TEMPLATE_TEST_CASE_METHOD( Fixture,
+                           "Optional sub-archive in describe even when uninitialized ",
+                           "[GeneratedArchive]",
+                           std::true_type,
+                           std::false_type )
+{
+    auto outer_builder = OuterWithOptionalBuilder::open( Fixture< TestType >::storage );
+    CHECK( outer_builder );
+
+    flatdata::Struct< AStruct > o;
+    ( *o ).value = 17u;
+    outer_builder.set_outer( *o );
+
+    auto outer = OuterWithOptional::open( Fixture< TestType >::storage );
+    REQUIRE( outer.is_open( ) );
+    REQUIRE( outer.outer( ).value == 17u );
+
+    const char* const expected
+        = R"data(================================================================================
+Flatdata Archive: OuterWithOptional
+================================================================================
+Resource                             Optional  Too Large  Loaded    Details
+================================================================================
+outer                                NO        NO         YES       Structure of size 1
+archive_resource                     YES       NO         NO        FATAL: Resource storage not initialized. Please check archive path.
```

Comment:
I guess the FATAL should be removed, since it is not fatal for optional subarchives

Same if true for `check archive path`.

Can you rephrase it into a neutral message? Or have a different message for root and nested?